src/Features/Core/Portable/UseCollectionInitializer/ObjectCreationExpressionAnalyzer.cs

```diff
@@ -186,14 +186,15 @@ private void AddMatches(ArrayBuilder<TExpressionStatementSyntax> matches)
                 return false;
             }
 
-            _syntaxFacts.GetPartsOfMemberAccessExpression(memberAccess, out instance, out var memberName);
+            _syntaxFacts.GetPartsOfMemberAccessExpression(memberAccess, out var localInstance, out var memberName);
             _syntaxFacts.GetNameAndArityOfSimpleName(memberName, out var name, out var arity);
 
             if (arity != 0 || !name.Equals(nameof(IList.Add)))
             {
                 return false;
             }
 
+            instance = localInstance;
```

Comment:
we were setting the value, but returning 'false'.  the caller used both the return value and the out value.  The intention was that if 'false' was returned, then null should be the out-val.